api/invite.go

```diff
@@ -75,6 +82,82 @@ func (a *Api) checkForDuplicateInvite(ctx context.Context, inviteeEmail, invitor
 
+//Checks do they have an existing invite or are they already a team member
+//Or are they an existing user and already in the group?
+func (a *Api) checkForDuplicateTeamInvite(ctx context.Context, inviteeEmail, invitorID, token string, team store.Team, invite models.Type, res http.ResponseWriter) (bool, *shoreline.UserData) {
+
+	//already has invite from this user?
+	invites, _ := a.Store.FindConfirmations(
+		ctx,
+		&models.Confirmation{
+			Email:  inviteeEmail,
+			TeamID: team.ID,
+			Type:   invite},
+		[]models.Status{models.StatusPending},
+		[]models.Type{},
+	)
+
+	if len(invites) > 0 {
+		//rule is we cannot send if the invite is not yet expired
+		if !invites[0].IsExpired() {
+			log.Println(statusExistingInviteMessage)
+			log.Println("last invite not yet expired")
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusConflict, statusExistingInviteMessage)}
+			a.sendModelAsResWithStatus(res, statusErr, http.StatusConflict)
+			return true, nil
+		}
+	}
+
+	invitedUsr := a.findExistingUser(inviteeEmail, a.sl.TokenProvide())
+	// call the teams service to check if the user is already a member
+	if invitedUsr != nil {
+		if isMember := a.isTeamMember(invitedUsr.UserID, team, false); isMember {
+			log.Printf("checkForDuplicateTeamInvite: invited [%s] user is already a member of [%s]", inviteeEmail, team.Name)
+			statusErr := &status.StatusError{Status: status.NewStatus(http.StatusConflict, statusExistingMemberMessage)}
+			a.sendModelAsResWithStatus(res, statusErr, http.StatusConflict)
+			return true, invitedUsr
+		}
+		return false, invitedUsr
+	}
+	return false, nil
+}
+
+func (a *Api) getUserPreferences(userid string, res http.ResponseWriter) *models.Preferences {
+	// let's get the invitee user preferences
+	inviteePreferences := &models.Preferences{}
+	if err := a.seagull.GetCollection(userid, "preferences", a.sl.TokenProvide(), inviteePreferences); err != nil {
+		a.sendError(
+			res,
+			http.StatusInternalServerError,
+			STATUS_ERR_FINDING_USR,
+			"send invitation: error getting invitee user preferences: ",
+			err.Error())
+	}
+	return inviteePreferences
+}
+
+func (a *Api) getUserLanguage(userid string, res http.ResponseWriter) string {
+	// let's get the invitee user preferences
+	language := "en"
+	inviteePreferences := a.getUserPreferences(userid, res)
+	// does the invitee have a preferred language?
+	if inviteePreferences.DisplayLanguage != "" {
+		language = inviteePreferences.DisplayLanguage
+	}
+	return language
+}
+
+func (a *Api) isTeamAdmin(userid string, team store.Team) (bool, error) {
+	for j := 0; j < len(team.Members); j++ {
+		if team.Members[j].UserID == userid {
+			if team.Members[j].Role == "admin" {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
 // @Summary Get list of received invitations for logged-in user
 // @Description  Get list of received invitations that have been sent to this user but not yet acted upon.
 // @ID hydrophone-api-GetReceivedInvitations
```

Comment:
Could you add this line below to the documentation to fix swagger documentation ? 
`// @Success 200 {array} models.Confirmation`

instead of // @Success 200 {string} string "OK" (line 167)